validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java

```diff
@@ -205,38 +219,51 @@ public SafeFuture<Void> subscribeToPersistentSubnets(
   public SafeFuture<List<SubmitDataError>> sendSignedAttestations(List<Attestation> attestations) {
     return relayRequest(
         apiChannel -> apiChannel.sendSignedAttestations(attestations),
-        BeaconNodeRequestLabels.PUBLISH_ATTESTATION_METHOD);
+        BeaconNodeRequestLabels.PUBLISH_ATTESTATION_METHOD,
+        failoversSendSignedDuties);
   }
 
   @Override
   public SafeFuture<List<SubmitDataError>> sendAggregateAndProofs(
       final List<SignedAggregateAndProof> aggregateAndProofs) {
     return relayRequest(
         apiChannel -> apiChannel.sendAggregateAndProofs(aggregateAndProofs),
-        BeaconNodeRequestLabels.PUBLISH_AGGREGATE_AND_PROOFS_METHOD);
+        BeaconNodeRequestLabels.PUBLISH_AGGREGATE_AND_PROOFS_METHOD,
+        failoversSendSignedDuties);
   }
 
   @Override
   public SafeFuture<SendSignedBlockResult> sendSignedBlock(final SignedBeaconBlock block) {
+    final UInt64 slot = block.getMessage().getSlot();
+    if (block.isBlinded() && blindedBlockCreatorCache.containsKey(slot)) {
+      final ValidatorApiChannel blockCreatorApiChannel = blindedBlockCreatorCache.remove(slot);
```

Comment:
Am I right here saying that we may leak the slot in the map if we never end up calling send? better using and LRUCache?